### ISSUE TITLE
Adding new field in existing models 

### ISSUE THREAD

**Issue by vesaalexandru95** [django-tenants/django-tenants#708]
Hello !
How should be done the migration for every tenant when I add a new field in existing models?
I've tried with --makemigrations for the specific app and after that migrate at every tenant but nothing changed.

Thanks a lot,
Alex

**Comment by Dresdn** [comment]
I'm assuming the app is specified in your `TENANT_APPS` section. Once you create your migrations via `./manage.py --makemigrations`, you can run `./manage.py migrate` which goes through and automagically performs a migrate for each tenant.

If you're having issues, share some output and configs to help diagnose.

**Comment by Dresdn** [comment]
Can you provide more detail @Alihassanc5? What your `TENANT_APPS` and `SHARED_APPS` look like, the app and model you're working with, output, etc.

Not much to go on to provide any assistance.

**Comment by Alihassanc5** [comment]
@Dresdn that's how my apps looks like:

```
CUSTOM_APPS = [
    "aurora_research.core",
    "aurora_research.users",
    "aurora_research.researches",
    "aurora_research.payments",
]

LOCAL_TENANT_APP = ["aurora_research.tenants"]

SHARED_APPS = [
    "django_tenants",
    "django.contrib.auth",
    "django.contrib.contenttypes",
    "django.contrib.sessions",
    "django.contrib.sites",
    "django.contrib.messages",
    "django.contrib.staticfiles",
    "django.contrib.admin",
    "django.forms",
] + LOCAL_TENANT_APP

TENANT_APPS = CUSTOM_APPS

INSTALLED_APPS = DEFAULT_APPS + THIRD_PARTY_APPS + CUSTOM_APPS + LOCAL_TENANT_APP
```

Note: I have integrated django-tenants to my existing db, all the migrations were applied perfectly first time, then I have added a new field, the migration was applied but there was no change in DB, and I was getting Programming Error "new_field" does not exist in table.

**Comment by Dresdn** [comment]
Since you didn't specify, I'm assuming you added a field to a model in one of the `aurora_research` apps. If so, it is expected that these apps wouldn't be available in the Public tenant as the Public tenant only consists of apps from the `SHARED_APPS`. Each tenant gets it's own tables in the schema for all the `TENANT_APPS`, while the "public" apps have their tables create in the `public` tenant (via `SHARED_APPS`).

Also, why the complex `*_APPS` settings? You only really need two: `SHARED_APPS` and `TENANT_APPS`.

**Comment by Alihassanc5** [comment]
@Dresdn Thank you for your response. Yes, I am adding my field to tenant-specific models and expecting this to be migrated to the `public` schema tenant as well. What, then, is the purpose of the public schema? Shouldn't all tenants, including the public schema, have tenant-specific apps? If code changes occur in my tenant-specific app, it may cause programming errors and prevent access to my public schema tenant. Should I run migrations on an empty database? If so, how will tenant users function? I have included `django.contrib.admin` in my shared_apps (as recommended by django_tenants) and users.User in Tenant_apps to create users tenant-specific.

**Comment by Dresdn** [comment]
> What, then, is the purpose of the public schema? Shouldn't all tenants, including the public schema, have tenant-specific apps?

No, the public schema should not have any tenant-specific apps. I suggest you read https://django-tenants.readthedocs.io/en/latest/#shared-applications to understand the difference between `public` and `tenant` schemas. That may answer a few of your questions.

> Should I run migrations on an empty database?

Migrations should be treated no differently than a non-tenant Django application.

> If so, how will tenant users function?

For users, there are a lot of challenges with `django-tenants` if you want to share users across tenants, like permissions. For this reason, https://github.com/Corvia/django-tenant-users (note, I'm a maintainer there) was created to help with that and various other things.

**Comment by Alihassanc5** [comment]
@Dresdn Thank you so much for clearing these things :), django-tenant-users seems like a library to handle users across the tenants, but my usecase is different, I don't want a user to share across tenants. Our users will be tenant-specific that's the challenge i am facing right now, I placed `django.contrib.admin` in `SHARED_APPS` and  `aurora_research.users` in `TENANT_APPS` but when I start to migrate it gives me an error users_user not exists because it's in tenant_apps, Should I still need this library to handle this case? Or any way to handle this case using `django_tenants` app?

**Comment by tomturner** [comment]
Closing as part of a backlog tidy-up. This is a usage/integration question rather than a bug in django-tenants, and it has been inactive for some years — the versions and Django releases involved have moved on a long way since.

If you're still stuck, **[GitHub Discussions](https://github.com/django-tenants/django-tenants/discussions)** is the better home for questions like this and gets more eyes than the issue tracker.

And if you believe this *is* a bug in django-tenants, please reopen or open a fresh issue with the django-tenants and Django versions, a minimal `settings.py`, and the steps to reproduce — that's what makes it actionable.

Thanks for taking the time to write it up.
